src/StreamFactory.php

```diff
@@ -2,16 +2,25 @@
 
 namespace Http\Factory\Diactoros;
 
-use Psr\Http\Factory\StreamFactoryInterface;
+use Psr\Http\Factory\Stream\CallbackStreamFactoryInterface;
+use Psr\Http\Factory\Stream\ResourceStreamFactoryInterface;
+use Psr\Http\Factory\Stream\StreamFactoryInterface;
 use Zend\Diactoros\Stream;
 use Zend\Diactoros\CallbackStream;
 
-class StreamFactory implements StreamFactoryInterface
+class StreamFactory implements
+    CallbackStreamFactoryInterface,
+    ResourceStreamFactoryInterface,
+    StreamFactoryInterface
 {
-    public function createStream()
+    public function createStream($body = '')
```

Comment:
Why not make use of the diactoros stream itself ? I am looking at https://github.com/zendframework/zend-diactoros/blob/b4e7758556c97b5bb9a5260d898e9788ee800538/src/Stream.php#L32